File: loops/loop_notes.py
#Ryan Crop, Loop notes python
# 1 What is a loop? 
    # A section of code that is going to repeat.
# 2 What are the two types of loops?
    #While loop
x = 0
while x < 10:
    print("Hello", x)
    x += 1
    #for loop
nums = [3,5,7,2,8]
for num in nums:
    print(num)
# 3 What is iteration
    # The act of repeating
# 4 What are lists? 
    # A bunch of values in the same variable.
familys = ["Reese", "caden", "julia", "dad", "mom"]
#Print one item from the list
print(familys[2])
#change an item in a list
familys[2] = "Jules"
#remove an item from list
familys.pop(4)
# 5 How do you make lists in python? family = ["Reese", "caden", "julia", "dad", "mom"]
# 6 How do you make for loops in python? 
#proper list printing with a for loop
for family in familys:
    print(f"{num}. {family} Crop")
    num += 1

for num in range (1, 11, 3):
    #             num, num, How much it goes up by 
    print(num)
# 7 How do you make while loops in python?
import random

num = 1
rand = random.randint(1,20)
while num < 21:
    if num == rand:
        print("Goose!")
        break # Stops until finished
    else:
        print("Duck")
    num += 1

# continue tells the loop to stop that iteration and start over

File: loops/loop_notes.c
//Ryan Crop, Loop notes c
#include <stdio.h>

int main(void){
//What is a loop?  
    // a section of code that repates
//What are the two types of loops? 
     // For
//int x;
//for (x=0; x<10; x++){
    //printf("Hello");
//}
    //While
//int i= 1;
//while(i<10){
    //printf("%d\n", i);
    //i++;
//}
    //What is iteration 
        //The act of repeating
    //What are lists (Array)?
        // A bunch of values in one variable
    //How do you make lists (Arrays) in C?
int grades[] = {97, 95, 100, 94, 81, 96, 99};
    // In the brakets we say how long the list will be, if the list is set there brackets can be empty
    //data type is whatever is in the bracket

printf("%d\n", grades[3]); // To print one item we put the index number in the brackets when we print.
grades[2] = 73; // udapte grades one at a time using the index number
printf("%d\n", grades[2]);
//This tells me the number of bytes in it
printf("%lu", sizeof(grades));
//How to get the size of the array
int length = sizeof(grades)/sizeof(grades[0]);
printf("%d\n", length);

    //How do you make for loops in C?
    //Iterators should be x or i
int t;
for(t=0;t<=10;t+=2){
    printf("%d\n", t);
}
int l;
for(l=0; l< length; l++){
    printf("%d\n", grades[l]);
}
    //How do you make while loops in C?
int iterator= 100;
while(iterator>= 0){
    printf("%d\n", iterator);
    iterator -= 10;

}

char movies[][20] = {"Cinderella", "The Smerf Movie", "Transformers", "Cars", "Up", "1984"};
int mlength = sizeof(movies)/sizeof(movies[0]);
int m = 0;
while(m<mlength){
    printf("I like the movie %s\n", movies[m]);
    m++;
}

    return 0;
}